sklearn/linear_model/tests/test_ridge.py

```diff
@@ -618,3 +619,55 @@ def test_sparse_cg_max_iter():
     reg = Ridge(solver="sparse_cg", max_iter=1)
     reg.fit(X_diabetes, y_diabetes)
     assert_equal(reg.coef_.shape[0], X_diabetes.shape[1])
+
+
+def make_noisy_forward_data(
+    n_samples=100,
+    n_features=200,
+    n_targets=10,
+    train_frac=.8,
+    noise_levels=None,
+    random_state=42):
+    """Creates a simple, dense, noisy forward linear model with multiple
+    output."""
+    rng = check_random_state(random_state)
+    n_train = int(train_frac * n_samples)
+    train = slice(None, n_train)
+    test = slice(n_train, None)
+    X = rng.randn(n_samples, n_features)
+    W = rng.randn(n_features, n_targets)
+    Y_clean = X.dot(W)
+    if noise_levels is None:
+        noise_levels = rng.randn(n_targets) ** 2
+    noise_levels = np.atleast_1d(noise_levels) * np.ones(n_targets)
+    noise = rng.randn(*Y_clean.shape) * noise_levels * Y_clean.std(0)
+    Y = Y_clean + noise
+    return X, Y, W, train, test
+
+
+def test_ridge_gcv_with_sample_weights():
+
+    n_samples, n_features, n_targets = 20, 5, 7
+    X, Y, W, _, _ = make_noisy_forward_data(n_samples, n_features, n_targets)
```

Comment:
indeed, sorry. i copied this in from an ages old pr where i didn't know
about that ;)

will update.

On Thursday, April 2, 2015, Vlad Niculae <notifications@github.com> wrote:

> In sklearn/linear_model/tests/test_ridge.py
> <https://github.com/scikit-learn/scikit-learn/pull/4490#discussion_r27685564>
> :
>
> > +    test = slice(n_train, None)
> > +    X = rng.randn(n_samples, n_features)
> > +    W = rng.randn(n_features, n_targets)
> > +    Y_clean = X.dot(W)
> > +    if noise_levels is None:
> > +        noise_levels = rng.randn(n_targets) ** 2
> > +    noise_levels = np.atleast_1d(noise_levels) * np.ones(n_targets)
> > +    noise = rng.randn(*Y_clean.shape) * noise_levels * Y_clean.std(0)
> > +    Y = Y_clean + noise
> > +    return X, Y, W, train, test
> > +
> > +
> > +def test_ridge_gcv_with_sample_weights():
> > +
> > +    n_samples, n_features, n_targets = 20, 5, 7
> > +    X, Y, W, _, _ = make_noisy_forward_data(n_samples, n_features, n_targets)
>
> Couldn't this use sklearn.datasets.make_regression?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4490/files#r27685564>.
>